File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        maxWater = 0
        i, j = 0, len(height) - 1

        while i < j:
            area = (j - i) * min(height[i], height[j])
            maxWater = max(maxWater, area)
            if height[i] > height[j]:
                j -= 1
            else:
                i += 1
        return maxWater

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int water = 0;

        int l =0; 
        int r = height.length-1;
        int currentWater = 0;
        while(l < r){
            if(height[l] < height[r]){
                currentWater = height[l] * (r-l);
                l++;
            }else{
                currentWater = height[r] * (r - l);
                r--;
            }
            water = Math.max(currentWater,water);
        }
        return water;
    }
}